src/compat/dbtype_def.h

```diff
@@ -506,7 +506,7 @@ extern "C"
 /* Maximum allowable class name. */
 #define DB_MAX_CLASS_LENGTH (DB_MAX_IDENTIFIER_LENGTH-DB_MAX_SCHEMA_LENGTH-4)
 
-#define DB_MAX_SPEC_LENGTH       4096
+#define DB_MAX_SPEC_LENGTH       (0x3FFFFFFF)
```

Comment:
How about '#define DB_MAX_SPEC_LENGTH DB_MAX_STRING_LENGTH'

Reply:
Actually DB_MAX_SPEC_LENGTH and DB_MAX_STRING_LENGTH are conceptually separated definition. firstly the designer of this header file 'dbtype_def.h' wanted to different length set for both values and I want to consider that thought. It's not problem to apply your suggestion to the code (now 'spec' attribute of the catalog class already faced to the limit of DB_MAX_VARCHAR_PRECISION) but if it's ok not to link it to DB_MAX_STRING_LENGTH, I think it just ok itself.